.github/workflows/create_sbom_report.yml

```
@@ -0,0 +1,119 @@
+name: Create and upload a SBOM to release assets
+# Inherited variables:
+# github.event.client_payload.imageLabel - AzDO image/OS label
+# github.event.client_payload.GHreleaseID - Current release ID
+# github.event.client_payload.imageVersion - AzDO image/OS version "major.minor"
+# 
+# Current SYFT tool issues:
+# macOS (minor): very long cataloging process (https://github.com/anchore/syft/issues/1328), 
+# macOS (major): promt privilegies that blocking process indefinetely (https://github.com/anchore/syft/issues/1367)
+# Windows (major): process fails on some symlinks (https://github.com/anchore/syft/issues/1094)
+on:
+  repository_dispatch:
+    types: [generate-sbom]
+jobs:
+  #Converting image OS variable for the next steps
+  initialize: 
+    strategy:
+      matrix: #Couples AzDO label <-> GHA image spec
+        include:
+          - label: ubuntu22
+            os: ubuntu-22.04
+          - label: ubuntu20
+            os: ubuntu-20.04
+          - label: ubuntu18
+            os: ubuntu-18.04
+          - label: macOS-12
+            os: macos-12
+          - label: macOS-11
+            os: macos-11
+          - label: macOS-10.15
+            os: macos-10.15
+          - label: win22
+            os: windows-2022
+          - label: win19
+            os: windows-2019
+    runs-on: ubuntu-latest
+    steps:
+      - name: Convert image label variable
+        run: if ("${{ github.event.client_payload.imageLabel }}"" -eq "{{ $matrix.label }}"") {echo "current-os=${{ $matrix.os }}" >> $GITHUB_OUTPUT}
+  #Checking image version on available runner
+  version-check:
+    strategy:
+      matrix:
+        include:
+          - os: Windows
+            path: C:\
+          - os: macOS
+            path: /Users/runner/
+          - os: Linux
+            path: /imagegeneration/
+    defaults:
+      run:
+        shell: pwsh
+    needs: initialize
+    runs-on: ${{ needs.initialize.output.current-os }}
+    steps:
+      - name: Available image version check
+        run: |
+          if ("${{ runner.os }}" -eq "${{ matrix.os }}") {
+            if ((Get-Content ${{ matrix.path }}imagedata.json | jq '.[1] | .detail') -match '\d{8}\.+\d') { 
+              if ($Matches[0] -eq ${{ github.event.client_payload.imageLabel }}) {
+                echo "Current runner use ${{ github.event.client_payload.imageLabel }} image version."
+              } else {
+                echo "Error. Current runner image version don't match ${{ github.event.client_payload.imageLabel }} image version."
+                exit 1
+              }
+            } else {
+              echo "Cannot parse image version."
+              exit 1
+            }
+          }
```

nit: swap if branches to avoid nested code and improve readability:

```suggestion
          if ("${{ runner.os }}" -ne "${{ matrix.os }}") {
            exit 0;
          }
          
          if (-not ((Get-Content ${{ matrix.path }}imagedata.json | jq '.[1] | .detail') -match '\d{8}\.+\d')) { 
              echo "Cannot parse image version."
              exit 1
          }
          
          if ($Matches[0] -eq ${{ github.event.client_payload.imageLabel }}) {
                echo "Current runner use ${{ github.event.client_payload.imageLabel }} image version."
          } else {
                echo "Error. Current runner image version don't match ${{ github.event.client_payload.imageLabel }} image version."
                exit 1
          }
```

---

Deleted part